### ISSUE TITLE
[Kafka] Allow Json consume to natively read AVRO Message

### ISSUE THREAD

**Issue by M3lkior** [authorjapps/zerocode#488]
# Context

For my organization project, i needed to produce and consume AVRO messages from  the Schema Registry so i develop a custom `KakfaClient` to produce the AVRO message.

Concerning the consumption, i detected after a debugging phase that the actual `KafkaReceiver` is almost able to treat an incoming AVRO message format  thanks to the `readJson` method

Here is the actual implementation 

```
            ConsumerJsonRecord jsonRecord = new ConsumerJsonRecord(thisRecord.key(), null, valueNode, headersMap);
```

Here is the code which allow parsing AVRO record 

```
            jsonRecords.add(new ConsumerJsonRecord(keyNode, null, valueNode, headersMap));

```
with `keyNode` 
```
            JsonNode keyNode = objectMapper.readTree(key.toString());

```

I can do a PR if this issue is accepted

**Comment by nirmalchandra** [comment]
Thanks @M3lkior. This looks great.
Yes please, please go ahead with the PR and assign the following folks as reviewers please.
@sparrowV , @respondsid , @dandalavinod, @ddingel, @santhoshTpixler 

Will assign additional reviewers if needed.

**Comment by nirmalchandra** [comment]
Integration test log evidence of PASS:

```JSON

2021-05-14 12:33:03,362 
--------- TEST-STEP-CORRELATION-ID: 754c28e2-a528-48be-9619-5511a47a9aaa ---------
*requestTimeStamp:2021-05-14T12:33:02.077
step:register_avro_schema_value_step
id:null
url:http://localhost:8081/subjects/myavrorecord/versions
method:POST
request:
{
  "headers" : {
    "Content-Type" : "application/vnd.schemaregistry.v1+json",
    "Accept" : "application/vnd.schemaregistry.v1+json, application/vnd.schemaregistry+json, application/json"
  },
  "body" : {
    "schema" : "{\"type\":\"record\",\"name\":\"myavrorecord\",\"fields\":[{\"name\":\"avrof1\",\"type\":\"string\"}]}"
  }
} 
--------- TEST-STEP-CORRELATION-ID: 754c28e2-a528-48be-9619-5511a47a9aaa ---------
Response:
{
  "status" : 200,
  "headers" : {
    "Date" : [ "Fri, 14 May 2021 11:33:02 GMT" ],
    "Content-Length" : [ "8" ],
    "Vary" : [ "Accept-Encoding, User-Agent" ],
    "Content-Type" : [ "application/vnd.schemaregistry.v1+json" ]
  },
  "body" : {
    "id" : 1
  }
}
*responseTimeStamp:2021-05-14T12:33:03.354 
*Response delay:1277.0 milli-secs 
---------> Expected Response: <----------
{
  "status" : 200,
  "body" : {
    "id" : 1
  }
} 
 
-done-
 
--------- TEST-STEP-CORRELATION-ID: e3992c32-4873-434c-8c3d-40fea73ad600 ---------
*requestTimeStamp:2021-05-14T12:33:03.364
step:register_avro_schema_key_step
id:null
url:http://localhost:8081/subjects/myavrorecordkey/versions
method:POST
request:
{
  "headers" : {
    "Content-Type" : "application/vnd.schemaregistry.v1+json",
    "Accept" : "application/vnd.schemaregistry.v1+json, application/vnd.schemaregistry+json, application/json"
  },
  "body" : {
    "schema" : "{\"type\":\"record\",\"name\":\"myavrorecordkey\",\"fields\":[{\"name\":\"key\",\"type\":\"string\"}]}"
  }
} 
--------- TEST-STEP-CORRELATION-ID: e3992c32-4873-434c-8c3d-40fea73ad600 ---------
Response:
{
  "status" : 200,
  "headers" : {
    "Date" : [ "Fri, 14 May 2021 11:33:03 GMT" ],
    "Content-Length" : [ "8" ],
    "Vary" : [ "Accept-Encoding, User-Agent" ],
    "Content-Type" : [ "application/vnd.schemaregistry.v1+json" ]
  },
  "body" : {
    "id" : 2
  }
}
*responseTimeStamp:2021-05-14T12:33:03.389 
*Response delay:25.0 milli-secs 
---------> Expected Response: <----------
{
  "status" : 200,
  "body" : {
    "id" : 2
  }
} 
 
-done-
  
--------- TEST-STEP-CORRELATION-ID: 71b34d33-954b-4f7e-abfe-573332aa28d8 ---------
*requestTimeStamp:2021-05-14T12:33:03.392
step:produce_avro_msg_without_key_step
id:null
url:http://localhost:8082/topics/demo-avro-1
method:POST
request:
{
  "headers" : {
    "Content-Type" : "application/vnd.kafka.avro.v2+json",
    "Accept" : "application/vnd.kafka.v2+json"
  },
  "body" : {
    "value_schema_id" : "1",
    "records" : [ {
      "key" : null,
      "value" : {
        "avrof1" : "it works"
      }
    } ]
  }
} 
--------- TEST-STEP-CORRELATION-ID: 71b34d33-954b-4f7e-abfe-573332aa28d8 ---------
Response:
{
  "status" : 200,
  "headers" : {
    "Date" : [ "Fri, 14 May 2021 11:33:03 GMT" ],
    "Vary" : [ "Accept-Encoding, User-Agent" ],
    "Content-Type" : [ "application/vnd.kafka.v2+json" ]
  },
  "body" : {
    "offsets" : [ {
      "partition" : 0,
      "offset" : 0,
      "error_code" : null,
      "error" : null
    } ],
    "key_schema_id" : null,
    "value_schema_id" : 1
  }
}
*responseTimeStamp:2021-05-14T12:33:04.850 
*Response delay:1458.0 milli-secs 
---------> Expected Response: <----------
{
  "status" : 200,
  "body" : {
    "offsets" : [ {
      "partition" : 0,
      "offset" : "$NOT.NULL",
      "error_code" : null,
      "error" : null
    } ]
  }
} 
 
-done-

--------- TEST-STEP-CORRELATION-ID: dd980b58-3c65-4ff1-a574-025d0bfa35a6 ---------
*requestTimeStamp:2021-05-14T12:33:04.854
step:consume_avro_msg_without_key_as_avro
id:null
url:kafka-topic:demo-avro-1
method:CONSUME
request:
{
  "consumerLocalConfigs" : {
    "recordType" : "AVRO",
    "commitSync" : true,
    "showRecordsConsumed" : true,
    "maxNoOfRetryPollsOrTimeouts" : 3
  }
} 
--------- TEST-STEP-CORRELATION-ID: dd980b58-3c65-4ff1-a574-025d0bfa35a6 ---------
Response:
{
  "records" : [ {
    "key" : null,
    "value" : {
      "avrof1" : "it works"
    },
    "headers" : { }
  } ],
  "size" : 1
}
*responseTimeStamp:2021-05-14T12:33:12.616 
*Response delay:7762.0 milli-secs 
---------> Expected Response: <----------
{
  "size" : 1,
  "records" : [ {
    "value" : {
      "avrof1" : "it works"
    }
  } ]
} 
 
-done-
 
--------- TEST-STEP-CORRELATION-ID: 74cfc1c8-6b49-4ec2-9ed2-70a400d6af0f ---------
*requestTimeStamp:2021-05-14T12:33:12.618
step:produce_avro_msg_with_key_step
id:null
url:http://localhost:8082/topics/demo-avro-1
method:POST
request:
{
  "headers" : {
    "Content-Type" : "application/vnd.kafka.avro.v2+json",
    "Accept" : "application/vnd.kafka.v2+json"
  },
  "body" : {
    "key_schema_id" : "2",
    "value_schema_id" : "1",
    "records" : [ {
      "key" : {
        "key" : "key works"
      },
      "value" : {
        "avrof1" : "it works"
      }
    } ]
  }
} 
--------- TEST-STEP-CORRELATION-ID: 74cfc1c8-6b49-4ec2-9ed2-70a400d6af0f ---------
Response:
{
  "status" : 200,
  "headers" : {
    "Date" : [ "Fri, 14 May 2021 11:33:12 GMT" ],
    "Vary" : [ "Accept-Encoding, User-Agent" ],
    "Content-Type" : [ "application/vnd.kafka.v2+json" ]
  },
  "body" : {
    "offsets" : [ {
      "partition" : 0,
      "offset" : 1,
      "error_code" : null,
      "error" : null
    } ],
    "key_schema_id" : 2,
    "value_schema_id" : 1
  }
}
*responseTimeStamp:2021-05-14T12:33:12.667 
*Response delay:49.0 milli-secs 
---------> Expected Response: <----------
{
  "status" : 200,
  "body" : {
    "offsets" : [ {
      "partition" : 0,
      "offset" : "$NOT.NULL",
      "error_code" : null,
      "error" : null
    } ]
  }
} 
 
-done-
 
--------- TEST-STEP-CORRELATION-ID: 818683a4-2f78-41ef-9a95-ea6da0d03fbf ---------
*requestTimeStamp:2021-05-14T12:33:12.669
step:consume_avro_msg_with_key_as_avro
id:null
url:kafka-topic:demo-avro-1
method:CONSUME
request:
{
  "consumerLocalConfigs" : {
    "recordType" : "AVRO",
    "commitSync" : true,
    "showRecordsConsumed" : true,
    "maxNoOfRetryPollsOrTimeouts" : 3
  }
} 
--------- TEST-STEP-CORRELATION-ID: 818683a4-2f78-41ef-9a95-ea6da0d03fbf ---------
Response:
{
  "records" : [ {
    "key" : {
      "key" : "key works"
    },
    "value" : {
      "avrof1" : "it works"
    },
    "headers" : { }
  } ],
  "size" : 1
}
*responseTimeStamp:2021-05-14T12:33:18.770 
*Response delay:6101.0 milli-secs 
---------> Expected Response: <----------
{
  "size" : 1,
  "records" : [ {
    "key" : {
      "key" : "key works"
    },
    "value" : {
      "avrof1" : "it works"
    }
  } ]
} 
 
-done-

2021-05-14 12:33:18,892
**FINISHED executing all Steps for [Produce a JSON message to a kafka topic] **.
Steps were:[
    register_avro_schema_value_step, 
    register_avro_schema_key_step, 
    produce_avro_msg_without_key_step, 
    consume_avro_msg_without_key_as_avro, 
    produce_avro_msg_with_key_step, 
    consume_avro_msg_with_key_as_avro
]

```

**Comment by nirmalchandra** [comment]
Release to maven central is pending(in-progress). Then this issue can be closed.

**Comment by nirmalchandra** [comment]
@M3lkior , Just checking, how urgent is this for your project? Can it wait a week or so for another PR(under review) ?
Also, if you could kindly accept the collaborator invitation.
Link: https://github.com/authorjapps/zerocode/invitations

**Comment by M3lkior** [comment]
> @M3lkior , Just checking, how urgent is this for your project? Can it wait a week or so for another PR(under review) ?
> Also, if you could kindly accept the collaborator invitation.
> Link: https://github.com/authorjapps/zerocode/invitations

Yeah, it is ok to wait a week before the merge ; no problem

**Comment by authorjapps** [comment]
Released.
Available in v [1.3.28](https://github.com/authorjapps/zerocode/releases/tag/zerocode-tdd-parent-1.3.28)
